src/server/tiering/op_manager.cc

```diff
@@ -112,26 +113,37 @@ void OpManager::ProcessRead(size_t offset, std::string_view value) {
     for (auto& cb : ko.callbacks)
       modified |= cb(&key_value);
 
-    // Report item as fetched only after all action were executed, pass whether it was modified
-    ReportFetched(Borrowed(ko.id), key_value, ko.segment, modified);
+    if (!ko.deleting)
+      ko.deleting |= ReportFetched(Borrowed(ko.id), key_value, ko.segment, modified || ko.deleting);
+
+    if (ko.deleting)
+      deleting_full |= ReportDelete(ko.segment);
   }
 
-  if (info->delete_requested)
+  if (deleting_full)
     storage_.MarkAsFree(info->segment);
   pending_reads_.erase(offset);
 }
 
-OpManager::EntryOps& OpManager::ReadOp::ForId(EntryId id, DiskSegment key_segment) {
+OpManager::EntryOps& OpManager::ReadOp::ForPart(DiskSegment key_segment, EntryId id) {
```

Comment:
I find  it a bit confusing that you have two functions with the same name, one of them adds something to the state and another one is "read-only"

Reply:
what does "ForPart" means actually? What is Part? :)

Reply:
Renamed to ForSegment 🤓